[Test][AutoDiff] Replace sed command with echo/cat for internal shell compatibility

emirariemir opened the pull request:
Refactored `AutoDiff` test files that contains `REQUIRES: shell` by replacing the line:
```
// RUN: sed -e 's/import Swift$/import Swift; import _Differentiation/' %t/tmp.sil > %t/tmp_fixed.sil
```
with `echo` and `cat` commands:
```
// RUN: echo "import _Differentiation" > %t/tmp_preamble
// RUN: cat %t/tmp_preamble %t/tmp.sil > %t/tmp_fixed.sil
```
to make it compatible with LLVM Lit internal shell.

Partially address https://github.com/swiftlang/swift/issues/84407

emirariemir committed:
[Test][AutoDiff] Replace sed with echo/cat in sil_differentiability_witness.sil file

emirariemir committed:
[Test][AutoDiff] Replace sed with echo/cat in differentiable_function_type.swift file

Convert `sed` command with `echo` and `cat` commands to achieve the same logic for compatibility with LLVM Lit internal shell.

emirariemir committed:
[Test][AutoDiff] Replace sed with echo/cat for Lit shell

Convert the existing `sed` commands in `AutoDiff` test files that use `REQUIRES: shell` into equivalent logic using `echo` and `cat` for compatibility with LLVM Lit’s internal shell.

asl commented on test/AutoDiff/SIL/Serialization/differentiable_function_type.swift:
```diff
@@ -4,13 +4,10 @@
 // https://github.com/apple/swift/issues/54526
 // Workaround because import declarations are not preserved in .sib files.
-// RUN: sed -e 's/import Swift$/import Swift; import _Differentiation/' %t/tmp.sil > %t/tmp_fixed.sil
+// RUN: echo "import _Differentiation" > %t/tmp_preamble
```
I do not recall all all restrictions of internal shell, does one of the following work instead?

```shell
cat <(echo "import _Differentiation") tmp.sil > tmp_fixed.sil
(echo "import _Differentiation"; cat tmp.sil) > tmp_fixed.sil
```

or even
```shell
echo "import _Differentiation" | cat - tmp.sil > tmp_fixed.sil
```

emirariemir replied:
I think that the `<(...)` syntax is only supported by BASH and not by internal shell, so that test would fail in Windows CI or shells other than Bash. I think keeping it as two atomic, standard commands is the most robust approach (thanks to @hnrklssn for suggesting this approach in the first place):
```
// RUN: echo "import _Differentiation" > %t/tmp_fixed.sil
// RUN: cat %t/tmp.sil >> %t/tmp_fixed.sil
```

hnrklssn commented:
@swift-ci please smoke test

emirariemir commented:
Oh no, Windows platform test failed 😭 ! I cannot see the cause in details, it says:

> No problems were identified. If you know why this problem occurred, please add a suitable Cause for it.

How do I identify the problem that makes the test fail?

emirariemir commented:
> If you click on the failed CI bot, and then go to "View as plain text" you can see the build and test log (where the above excerpt is from).

Thanks so much for the explanation! I’ll check the logs and try to fix it. If I run into issues, I’ll reach out again.

emirariemir committed:
[Test][AutoDiff] Make differentiability witness checks tolerant of Windows dllimport and dllexport

This change updates the affected `IRGEN` and `IRGEN-LABEL` checks to explicitly allow optional `dllimport` and `dllexport` attributes using regex patterns, while keeping the rest of the checks strict.

emirariemir commented:
> It looks like // IRGEN: @fooWJrSUUpSr = external global %swift.differentiability_witness, align [[PTR_ALIGNMENT:[0-9]+]] doesn't match @fooWJrSUUpSr = external dllimport global %swift.differentiability_witness, align 8 because of the extra dllimport on Windows. These tests were not running on Windows previously, since REQUIRES: shell is never fulfilled on Windows, so we'll have to make some small adjustments. Something like // IRGEN: @fooWJrSUUpSr = external {{(dllimport)?}} global %swift.differentiability_witness, align [[PTR_ALIGNMENT:[0-9]+]] should fix it.

I’ve updated the failing tests by adding optional `dllimport` and `dllexport` handling to the relevant FileCheck patterns, following the approach suggested by @hnrklssn. The changes are now up to date and ready for another review.

If you’d prefer `{{.*}}` approach, I can switch to that as well — happy to adjust based on feedback.

hnrklssn commented:
@swift-ci please smoke test windows platform

hnrklssn commented:
@swift-ci please test windows platform

asl commented:
> I may be blathering, but what happens when Windows reaches `cat`? Out of all the tests I've done, they've been platform specific (except [#84923](https://github.com/swiftlang/swift/pull/84923)), so I don't really know. But looking at the [Windows commands](https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/windows-commands), I don't see `cat`, but the specific test seems to pass? I know I'll work when we enable LIT test but does Windows already have that enabled?

`cat` is a built-in command in lit shell.

ramonasuncion commented:
Ah, yep. I ended up reading on the LLVM forum:

> However, lit also has an internal shell implementation, which is currently the default for Windows systems. --  https://discourse.llvm.org/t/rfc-enabling-the-lit-internal-shell-by-default/80179

hnrklssn commented:
@swift-ci please smoke test windows platform

hnrklssn commented:
> Oh! I now see a failed test instead of aborting:
> 
> ```
> 
> Failed Tests (1):
> 
>   Swift(windows-x86_64) :: Interop/Cxx/class/member-variables-typechecker.swift
> 
> ```
> 
> This file wasn’t part of the scope of this PR, so I’m not sure why it’s failing. I double-checked all the files touched by this PR and they seem to pass. Should I also look into `member-variables-typechecker.swift` as part of this PR, or could this be unrelated? @hnrklssn

It's probably unrelated, I'll see if I can find any known failures there. Here's the output for reference:

```

FAIL: Swift(windows-x86_64) :: Interop/Cxx/class/member-variables-typechecker.swift (4370 of 12243)
******************** TEST 'Swift(windows-x86_64) :: Interop/Cxx/class/member-variables-typechecker.swift' FAILED ********************
Exit Code: 1

Command Output (stdout):
--
# RUN: at line 1
't:\\5\\bin\\swift-frontend.exe' -target x86_64-unknown-windows-msvc  -module-cache-path T:\5\swift-test-results\x86_64-unknown-windows-msvc\clang-module-cache -vfsoverlay T:/5/tools/swift\stdlib\windows-vfs-overlay.yaml -strict-implicit-module-context -Xcc -Xclang -Xcc -fbuiltin-headers-in-system-modules -swift-version 4  -define-availability 'SwiftStdlib 9999:macOS 9999, iOS 9999, watchOS 9999, tvOS 9999' -define-availability 'StdlibDeploymentTarget 9999:macOS 9999, iOS 9999, watchOS 9999, tvOS 9999' -define-availability 'SwiftStdlib 5.0:macOS 10.14.4, iOS 12.2, watchOS 5.2, tvOS 12.2' -define-availability 'StdlibDeploymentTarget 5.0:macOS 10.14.4, iOS 12.2, watchOS 5.2, tvOS 12.2' -define-availability 'SwiftStdlib 5.1:macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0' -define-availability 'StdlibDeploymentTarget 5.1:macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0' -define-availability 'SwiftStdlib 5.2:macOS 10.15.4, iOS 13.4, watchOS 6.2, tvOS 13.4' -define-availability 'StdlibDeploymentTarget 5.2:macOS 10.15.4, iOS 13.4, watchOS 6.2, tvOS 13.4' -define-availability 'SwiftStdlib 5.3:macOS 11.0, iOS 14.0, watchOS 7.0, tvOS 14.0' -define-availability 'StdlibDeploymentTarget 5.3:macOS 11.0, iOS 14.0, watchOS 7.0, tvOS 14.0' -define-availability 'SwiftStdlib 5.4:macOS 11.3, iOS 14.5, watchOS 7.4, tvOS 14.5' -define-availability 'StdlibDeploymentTarget 5.4:macOS 11.3, iOS 14.5, watchOS 7.4, tvOS 14.5' -define-availability 'SwiftStdlib 5.5:macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0' -define-availability 'StdlibDeploymentTarget 5.5:macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0' -define-availability 'SwiftStdlib 5.6:macOS 12.3, iOS 15.4, watchOS 8.5, tvOS 15.4' -define-availability 'StdlibDeploymentTarget 5.6:macOS 12.3, iOS 15.4, watchOS 8.5, tvOS 15.4' -define-availability 'SwiftStdlib 5.7:macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0' -define-availability 'StdlibDeploymentTarget 5.7:macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0' -define-availability 'SwiftStdlib 5.8:macOS 13.3, iOS 16.4, watchOS 9.4, tvOS 16.4' -define-availability 'StdlibDeploymentTarget 5.8:macOS 13.3, iOS 16.4, watchOS 9.4, tvOS 16.4' -define-availability 'SwiftStdlib 5.9:macOS 14.0, iOS 17.0, watchOS 10.0, tvOS 17.0' -define-availability 'StdlibDeploymentTarget 5.9:macOS 14.0, iOS 17.0, watchOS 10.0, tvOS 17.0' -define-availability 'SwiftStdlib 5.10:macOS 14.4, iOS 17.4, watchOS 10.4, tvOS 17.4, visionOS 1.1' -define-availability 'StdlibDeploymentTarget 5.10:macOS 14.4, iOS 17.4, watchOS 10.4, tvOS 17.4, visionOS 1.1' -define-availability 'SwiftStdlib 6.0:macOS 15.0, iOS 18.0, watchOS 11.0, tvOS 18.0, visionOS 2.0' -define-availability 'StdlibDeploymentTarget 6.0:macOS 15.0, iOS 18.0, watchOS 11.0, tvOS 18.0, visionOS 2.0' -define-availability 'SwiftStdlib 6.1:macOS 15.4, iOS 18.4, watchOS 11.4, tvOS 18.4, visionOS 2.4' -define-availability 'StdlibDeploymentTarget 6.1:macOS 15.4, iOS 18.4, watchOS 11.4, tvOS 18.4, visionOS 2.4' -define-availability 'SwiftStdlib 6.2:macOS 26.0, iOS 26.0, watchOS 26.0, tvOS 26.0, visionOS 26.0' -define-availability 'StdlibDeploymentTarget 6.2:macOS 26.0, iOS 26.0, watchOS 26.0, tvOS 26.0, visionOS 26.0' -define-availability 'SwiftStdlib 6.3:macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, visionOS 9999' -define-availability 'StdlibDeploymentTarget 6.3:macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, visionOS 9999' -define-availability 'SwiftCompatibilitySpan 5.0:macOS 10.14.4, iOS 12.2, watchOS 5.2, tvOS 12.2, visionOS 1.0' -define-availability 'SwiftCompatibilitySpan 6.2:macOS 26.0, iOS 26.0, watchOS 26.0, tvOS 26.0, visionOS 26.0' -typo-correction-limit 10  -typecheck -verify -disable-objc-attr-requires-foundation-module C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class\member-variables-typechecker.swift -I C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs -enable-experimental-cxx-interop
# executed command: 't:\\5\\bin\\swift-frontend.exe' -target x86_64-unknown-windows-msvc -module-cache-path 'T:\5\swift-test-results\x86_64-unknown-windows-msvc\clang-module-cache' -vfsoverlay 'T:/5/tools/swift\stdlib\windows-vfs-overlay.yaml' -strict-implicit-module-context -Xcc -Xclang -Xcc -fbuiltin-headers-in-system-modules -swift-version 4 -define-availability 'SwiftStdlib 9999:macOS 9999, iOS 9999, watchOS 9999, tvOS 9999' -define-availability 'StdlibDeploymentTarget 9999:macOS 9999, iOS 9999, watchOS 9999, tvOS 9999' -define-availability 'SwiftStdlib 5.0:macOS 10.14.4, iOS 12.2, watchOS 5.2, tvOS 12.2' -define-availability 'StdlibDeploymentTarget 5.0:macOS 10.14.4, iOS 12.2, watchOS 5.2, tvOS 12.2' -define-availability 'SwiftStdlib 5.1:macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0' -define-availability 'StdlibDeploymentTarget 5.1:macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0' -define-availability 'SwiftStdlib 5.2:macOS 10.15.4, iOS 13.4, watchOS 6.2, tvOS 13.4' -define-availability 'StdlibDeploymentTarget 5.2:macOS 10.15.4, iOS 13.4, watchOS 6.2, tvOS 13.4' -define-availability 'SwiftStdlib 5.3:macOS 11.0, iOS 14.0, watchOS 7.0, tvOS 14.0' -define-availability 'StdlibDeploymentTarget 5.3:macOS 11.0, iOS 14.0, watchOS 7.0, tvOS 14.0' -define-availability 'SwiftStdlib 5.4:macOS 11.3, iOS 14.5, watchOS 7.4, tvOS 14.5' -define-availability 'StdlibDeploymentTarget 5.4:macOS 11.3, iOS 14.5, watchOS 7.4, tvOS 14.5' -define-availability 'SwiftStdlib 5.5:macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0' -define-availability 'StdlibDeploymentTarget 5.5:macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0' -define-availability 'SwiftStdlib 5.6:macOS 12.3, iOS 15.4, watchOS 8.5, tvOS 15.4' -define-availability 'StdlibDeploymentTarget 5.6:macOS 12.3, iOS 15.4, watchOS 8.5, tvOS 15.4' -define-availability 'SwiftStdlib 5.7:macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0' -define-availability 'StdlibDeploymentTarget 5.7:macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0' -define-availability 'SwiftStdlib 5.8:macOS 13.3, iOS 16.4, watchOS 9.4, tvOS 16.4' -define-availability 'StdlibDeploymentTarget 5.8:macOS 13.3, iOS 16.4, watchOS 9.4, tvOS 16.4' -define-availability 'SwiftStdlib 5.9:macOS 14.0, iOS 17.0, watchOS 10.0, tvOS 17.0' -define-availability 'StdlibDeploymentTarget 5.9:macOS 14.0, iOS 17.0, watchOS 10.0, tvOS 17.0' -define-availability 'SwiftStdlib 5.10:macOS 14.4, iOS 17.4, watchOS 10.4, tvOS 17.4, visionOS 1.1' -define-availability 'StdlibDeploymentTarget 5.10:macOS 14.4, iOS 17.4, watchOS 10.4, tvOS 17.4, visionOS 1.1' -define-availability 'SwiftStdlib 6.0:macOS 15.0, iOS 18.0, watchOS 11.0, tvOS 18.0, visionOS 2.0' -define-availability 'StdlibDeploymentTarget 6.0:macOS 15.0, iOS 18.0, watchOS 11.0, tvOS 18.0, visionOS 2.0' -define-availability 'SwiftStdlib 6.1:macOS 15.4, iOS 18.4, watchOS 11.4, tvOS 18.4, visionOS 2.4' -define-availability 'StdlibDeploymentTarget 6.1:macOS 15.4, iOS 18.4, watchOS 11.4, tvOS 18.4, visionOS 2.4' -define-availability 'SwiftStdlib 6.2:macOS 26.0, iOS 26.0, watchOS 26.0, tvOS 26.0, visionOS 26.0' -define-availability 'StdlibDeploymentTarget 6.2:macOS 26.0, iOS 26.0, watchOS 26.0, tvOS 26.0, visionOS 26.0' -define-availability 'SwiftStdlib 6.3:macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, visionOS 9999' -define-availability 'StdlibDeploymentTarget 6.3:macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, visionOS 9999' -define-availability 'SwiftCompatibilitySpan 5.0:macOS 10.14.4, iOS 12.2, watchOS 5.2, tvOS 12.2, visionOS 1.0' -define-availability 'SwiftCompatibilitySpan 6.2:macOS 26.0, iOS 26.0, watchOS 26.0, tvOS 26.0, visionOS 26.0' -typo-correction-limit 10 -typecheck -verify -disable-objc-attr-requires-foundation-module 'C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class\member-variables-typechecker.swift' -I 'C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs' -enable-experimental-cxx-interop
# .---command stderr------------
# | <unknown>:0: warning: the '-enable-experimental-cxx-interop' flag is deprecated; please pass '-cxx-interoperability-mode=' instead
# | <unknown>:0: note: Swift will maintain source compatibility for imported APIs based on the selected compatibility mode, so updating the Swift compiler will not change how APIs are imported
# | <module-includes>:1:10: error: unexpected note produced: in file included from <module-includes>:1:
# | #include "member-variables.h"
# |          ^
# | <module-includes>:1:10: note: file '<module-includes>' is not parsed for 'expected' statements. Use '-verify-additional-file <module-includes>' to enable, or '-verify-ignore-unrelated' to ignore diagnostics in this file
# | #include "member-variables.h"
# |          ^
# | C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h:20:5: error: unexpected warning produced: unknown attribute 'no_unique_address' ignored
# |   [[no_unique_address]] Empty b;
# |     ^
# | C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h:20:5: note: file 'C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h' is not parsed for 'expected' statements. Use '-verify-additional-file C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h' to enable, or '-verify-ignore-unrelated' to ignore diagnostics in this file
# |   [[no_unique_address]] Empty b;
# |     ^
# | <module-includes>:1:10: error: unexpected note produced: in file included from <module-includes>:1:
# | #include "member-variables.h"
# |          ^
# | <module-includes>:1:10: note: file '<module-includes>' is not parsed for 'expected' statements. Use '-verify-additional-file <module-includes>' to enable, or '-verify-ignore-unrelated' to ignore diagnostics in this file
# | #include "member-variables.h"
# |          ^
# | C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h:22:5: error: unexpected warning produced: unknown attribute 'no_unique_address' ignored
# |   [[no_unique_address]] Empty d;
# |     ^
# | C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h:22:5: note: file 'C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h' is not parsed for 'expected' statements. Use '-verify-additional-file C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h' to enable, or '-verify-ignore-unrelated' to ignore diagnostics in this file
# |   [[no_unique_address]] Empty d;
# |     ^
# | <module-includes>:1:10: error: unexpected note produced: in file included from <module-includes>:1:
# | #include "member-variables.h"
# |          ^
# | <module-includes>:1:10: note: file '<module-includes>' is not parsed for 'expected' statements. Use '-verify-additional-file <module-includes>' to enable, or '-verify-ignore-unrelated' to ignore diagnostics in this file
# | #include "member-variables.h"
# |          ^
# | C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h:24:5: error: unexpected warning produced: unknown attribute 'no_unique_address' ignored
# |   [[no_unique_address]] Empty f;
# |     ^
# | C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h:24:5: note: file 'C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h' is not parsed for 'expected' statements. Use '-verify-additional-file C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h' to enable, or '-verify-ignore-unrelated' to ignore diagnostics in this file
# |   [[no_unique_address]] Empty f;
# |     ^
# | <module-includes>:1:10: error: unexpected note produced: in file included from <module-includes>:1:
# | #include "member-variables.h"
# |          ^
# | <module-includes>:1:10: note: file '<module-includes>' is not parsed for 'expected' statements. Use '-verify-additional-file <module-includes>' to enable, or '-verify-ignore-unrelated' to ignore diagnostics in this file
# | #include "member-variables.h"
# |          ^
# | C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h:32:5: error: unexpected warning produced: unknown attribute 'no_unique_address' ignored
# |   [[no_unique_address]] std::optional<int> a = {2};
# |     ^
# | C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h:32:5: note: file 'C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h' is not parsed for 'expected' statements. Use '-verify-additional-file C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h' to enable, or '-verify-ignore-unrelated' to ignore diagnostics in this file
# |   [[no_unique_address]] std::optional<int> a = {2};
# |     ^
# | <module-includes>:1:10: error: unexpected note produced: in file included from <module-includes>:1:
# | #include "member-variables.h"
# |          ^
# | <module-includes>:1:10: note: file '<module-includes>' is not parsed for 'expected' statements. Use '-verify-additional-file <module-includes>' to enable, or '-verify-ignore-unrelated' to ignore diagnostics in this file
# | #include "member-variables.h"
# |          ^
# | C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h:43:5: error: unexpected warning produced: unknown attribute 'no_unique_address' ignored
# |   [[no_unique_address]] OptInt a;
# |     ^
# | C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h:43:5: note: file 'C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h' is not parsed for 'expected' statements. Use '-verify-additional-file C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h' to enable, or '-verify-ignore-unrelated' to ignore diagnostics in this file
# |   [[no_unique_address]] OptInt a;
# |     ^
# | <module-includes>:1:10: error: unexpected note produced: in file included from <module-includes>:1:
# | #include "member-variables.h"
# |          ^
# | <module-includes>:1:10: note: file '<module-includes>' is not parsed for 'expected' statements. Use '-verify-additional-file <module-includes>' to enable, or '-verify-ignore-unrelated' to ignore diagnostics in this file
# | #include "member-variables.h"
# |          ^
# | C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h:62:5: error: unexpected warning produced: unknown attribute 'no_unique_address' ignored
# |   [[no_unique_address]] NonStandardLayoutClass<int> a;
# |     ^
# | C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h:62:5: note: file 'C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h' is not parsed for 'expected' statements. Use '-verify-additional-file C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h' to enable, or '-verify-ignore-unrelated' to ignore diagnostics in this file
# |   [[no_unique_address]] NonStandardLayoutClass<int> a;
# |     ^
# | <module-includes>:1:10: error: unexpected note produced: in file included from <module-includes>:1:
# | #include "member-variables.h"
# |          ^
# | <module-includes>:1:10: note: file '<module-includes>' is not parsed for 'expected' statements. Use '-verify-additional-file <module-includes>' to enable, or '-verify-ignore-unrelated' to ignore diagnostics in this file
# | #include "member-variables.h"
# |          ^
# | C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h:72:5: error: unexpected warning produced: unknown attribute 'no_unique_address' ignored
# |   [[no_unique_address]] struct { private: T x; public: char pad_me; } a;
# |     ^
# | C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h:72:5: note: file 'C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h' is not parsed for 'expected' statements. Use '-verify-additional-file C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h' to enable, or '-verify-ignore-unrelated' to ignore diagnostics in this file
# |   [[no_unique_address]] struct { private: T x; public: char pad_me; } a;
# |     ^
# | <module-includes>:1:10: error: unexpected note produced: in file included from <module-includes>:1:
# | #include "member-variables.h"
# |          ^
# | <module-includes>:1:10: note: file '<module-includes>' is not parsed for 'expected' statements. Use '-verify-additional-file <module-includes>' to enable, or '-verify-ignore-unrelated' to ignore diagnostics in this file
# | #include "member-variables.h"
# |          ^
# | C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h:97:5: error: unexpected warning produced: unknown attribute 'no_unique_address' ignored
# |   [[no_unique_address]] EmptyNotImported p0;
# |     ^
# | C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h:97:5: note: file 'C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h' is not parsed for 'expected' statements. Use '-verify-additional-file C:\Users\swift-ci\jenkins\workspace\swift-PR-windows\swift\test\Interop\Cxx\class/Inputs\member-variables.h' to enable, or '-verify-ignore-unrelated' to ignore diagnostics in this file
# |   [[no_unique_address]] EmptyNotImported p0;
# |     ^
# `-----------------------------
# error: command failed with exit status: 1
```

hnrklssn commented:
@swift-ci please smoke test

hnrklssn commented:
Sorry about the delay on getting this merged!
